File: tags/2.0-Beta-01/trunk/mobileCloud/android/2_0/test-suite/test-sync-port/src/main/java/org/openmobster/core/mobileCloud/android/module/sync/TestOneWayClientSync.java
/******************************************************************************
 * OpenMobster                                                                *
 * Copyright 2008, OpenMobster, and individual                                *
 * contributors as indicated by the @authors tag. See the                     *
 * copyright.txt in the distribution for a full listing of                    *
 * individual contributors.                                                   *
 *                                                                            *
 * This is free software; you can redistribute it and/or modify it            *
 * under the terms of the GNU Lesser General Public License as                *
 * published by the Free Software Foundation; either version 2.1 of           *
 * the License, or (at your option) any later version.                        *
 *                                                                            *
 * This software is distributed in the hope that it will be useful,           *
 * but WITHOUT ANY WARRANTY; without even the implied warranty of             *
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU           *
 * Lesser General Public License for more details.                            *
 *                                                                            *
 * You should have received a copy of the GNU Lesser General Public           *
 * License along with this software; if not, write to the Free                *
 * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA         *
 * 02110-1301 USA, or see the FSF site: http://www.fsf.org.                   *
 ******************************************************************************/
package org.openmobster.core.mobileCloud.android.module.sync;


import org.openmobster.core.mobileCloud.android.module.mobileObject.MobileObject;


/**
 * 
 * @author openmobster@gmail.com
 *
 */
public final class TestOneWayClientSync extends AbstractSyncTest 
{		
	/**
	 * 
	 */
	public void runTest()
	{
		try
		{
			//Add test case
			this.setUp("add");
			SyncService.getInstance().performOneWayClientSync(service, service, false);					
			this.assertRecordAbsence("unique-1", "/TestOneWayClientSync/add");
			this.assertRecordAbsence("unique-2", "/TestOneWayClientSync/add");
			this.assertRecordPresence("unique-3", "/TestOneWayClientSync/add");
			this.assertRecordPresence("unique-4", "/TestOneWayClientSync/add");
			this.tearDown();
			
			//Replace test case
			this.setUp("replace");
			SyncService.getInstance().performOneWayClientSync(service, service, false);
			MobileObject afterUnique1 = this.getRecord("unique-1");
			MobileObject afterUnique2 = this.getRecord("unique-2");
			this.assertRecordPresence("unique-1", "/TestOneWayClientSync/replace");
			this.assertRecordPresence("unique-2", "/TestOneWayClientSync/replace");
			this.assertEquals(afterUnique1.getValue("message"), "<tag apos='apos' quote=\"quote\" ampersand='&'>unique-1/Message</tag>", 
			"/TestOneWayClientSync/replace/updated/unique-1");
			this.assertEquals(afterUnique2.getValue("message"), "<tag apos='apos' quote=\"quote\" ampersand='&'>unique-2/Updated/Client</tag>", 
			"/TestOneWayClientSync/replace/updated/unique-2");
			this.tearDown();
			
			//Delete test case
			this.setUp("delete");
			SyncService.getInstance().performOneWayClientSync(service, service, false);
			this.assertRecordPresence("unique-1", "/TestOneWayClientSync/delete");
			this.assertRecordAbsence("unique-2", "/TestOneWayClientSync/delete");
			this.tearDown();	
			
			//Conflict test case
			this.setUp("conflict");
			SyncService.getInstance().performOneWayClientSync(service, service, false);
			afterUnique1 = this.getRecord("unique-1");
			afterUnique2 = this.getRecord("unique-2");
			this.assertRecordPresence("unique-1", "/TestOneWayClientSync/conflict");
			this.assertRecordPresence("unique-2", "/TestOneWayClientSync/conflict");
			this.assertEquals(afterUnique1.getValue("message"), "<tag apos='apos' quote=\"quote\" ampersand='&'>unique-1/Updated/Client</tag>", 
			"/TestOneWayClientSync/conflict/updated/unique-1");
			this.assertEquals(afterUnique2.getValue("message"), "<tag apos='apos' quote=\"quote\" ampersand='&'>unique-2/Message</tag>", 
			"/TestOneWayClientSync/conflict/updated/unique-2");
			this.tearDown();
			
		}
		catch(Exception e)
		{
			e.printStackTrace(System.out);
			throw new RuntimeException(e.toString());
		}
	}	
}

File: tags/2.4-M3/trunk/iphone/mobilecloudlib/TestOneWayClientSync.m
/**
 * Copyright (c) {2003,2010} {openmobster@gmail.com} {individual contributors as indicated by the @authors tag}.
 * All rights reserved. This program and the accompanying materials
 * are made available under the terms of the Eclipse Public License v1.0
 * which accompanies this distribution, and is available at
 * http://www.eclipse.org/legal/epl-v10.html
 */

#import "TestOneWayClientSync.h"


/**
 * 
 * @author openmobster@gmail.com
 */
@implementation TestOneWayClientSync

+(id)withInit
{
	return [[[TestOneWayClientSync alloc] init] autorelease];
}

-(void) runTest
{
	SyncService *sync = [SyncService getInstance];
	
	//Add usecase
	[self setUp:@"add"];
	[sync performOneWayClientSync:channel :NO];
	[self assertRecordAbsence:@"unique-1" :@"/TestOneWayClientSync/add"];
	[self assertRecordAbsence:@"unique-2" :@"/TestOneWayClientSync/add"];
	[self assertRecordPresence:@"unique-3" :@"/TestOneWayClientSync/add"];
	[self assertRecordPresence:@"unique-4" :@"/TestOneWayClientSync/add"];
	[self tearDown];
	
	//Replace usecase
	[self setUp:@"replace"];
	[sync performOneWayClientSync:channel :NO];
	MobileObject *afterUnique1 = [self getRecord:@"unique-1"];
	MobileObject *afterUnique2 = [self getRecord:@"unique-2"];
	[self assertRecordPresence:@"unique-1" :@"/TestOneWayClientSync/replace"];
	[self assertRecordPresence:@"unique-2" :@"/TestOneWayClientSync/replace"];
	NSString *compareMessage1 = @"<tag apos='apos' quote=\"quote\" ampersand='&'>unique-1/Message</tag>";
	NSString *compareMessage2 = @"<tag apos='apos' quote=\"quote\" ampersand='&'>unique-2/Updated/Client</tag>";
	NSString *afterUnique1Msg = [afterUnique1 getValue:@"message"];
	NSString *afterUnique2Msg = [afterUnique2 getValue:@"message"];
	NSLog(@"%@",afterUnique1Msg);
	NSLog(@"%@",afterUnique2Msg);
	[self assertTrue:[afterUnique1Msg isEqualToString:compareMessage1] :@"/TestOneWayClientSync/replace/updated/unique-1"];
	[self assertTrue:[afterUnique2Msg isEqualToString:compareMessage2] :@"/TestOneWayClientSync/replace/updated/unique-2"];
	[self tearDown];
	
	//Delete usecase
	[self setUp:@"delete"];
	[sync performOneWayClientSync:channel :NO];
	[self assertRecordPresence:@"unique-1" :@"/TestOneWayClientSync/delete"];
	[self assertRecordAbsence:@"unique-2" :@"/TestOneWayClientSync/delete"];
	[self tearDown];
	
	//Conflict usecase
	[self setUp:@"conflict"];
	[sync performOneWayClientSync:channel :NO];
	afterUnique1 = [self getRecord:@"unique-1"];
	afterUnique2 = [self getRecord:@"unique-2"];
	[self assertRecordPresence:@"unique-1" :@"/TestOneWayClientSync/conflict"];
	[self assertRecordPresence:@"unique-2" :@"/TestOneWayClientSync/conflict"];
	compareMessage1 = @"<tag apos='apos' quote=\"quote\" ampersand='&'>unique-1/Updated/Client</tag>";
	compareMessage2 = @"<tag apos='apos' quote=\"quote\" ampersand='&'>unique-2/Message</tag>";
	afterUnique1Msg = [afterUnique1 getValue:@"message"];
	afterUnique2Msg = [afterUnique2 getValue:@"message"];
	NSLog(@"%@",afterUnique1Msg);
	NSLog(@"%@",afterUnique2Msg);
	[self assertTrue:[afterUnique1Msg isEqualToString:compareMessage1] :@"/TestOneWayClientSync/conflict/updated/unique-1"];
	[self assertTrue:[afterUnique2Msg isEqualToString:compareMessage2] :@"/TestOneWayClientSync/conflict/updated/unique-2"];
	[self tearDown];
}

-(NSString *) getInfo
{
	return [[self class] description];
}


@end
